### PR TITLE
[IMP] l10n_in_pos: receipt improve

### PR DESCRIPTION
**Pull request by jva-odoo**
task: https://www.odoo.com/web#id=2000231&action=327&model=project.task&view_type=form&menu_id=4720

pad: https://pad.odoo.com/p/r.d45a02d74c77f905291fef8abd30e90a

Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by pimodoo** [comment]
@jva-odoo Problem with this PR, not mergable and 27 commits

**Comment by jva-odoo** [comment]
@pimodoo Rebase is done...

File: addons/l10n_in_pos/static/src/js/receipt.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+        var line = _super_orderline.export_for_printing.apply(this,arguments);
+        line.l10n_in_hsn_code = this.get_product().l10n_in_hsn_code;
+        return line;
+    },
+});
+
+var _super_order = models.Order.prototype;
+models.Order = models.Order.extend({
+    export_for_printing: function () {
+        var receipt = _super_order.export_for_printing.apply(this,arguments);
```
Comment (pimodoo) [inline addons/l10n_in_pos/static/src/js/receipt.js]: In the point of sale module, you have 'client: client ? client.name : null ,' in the 'export_for_printing' function.
You can replace it by 'client: client ? client : null ,' and you won't have to do this override, and you will be able to use client.phone in the report.

P.S. check that client is not used and you should not replace it by 'client.name', but I think it is not used. You'll be able to set his name too as the specification requires it.

File: addons/l10n_in_pos/static/src/xml/pos_receipt.xml [hunk 2]
```diff
@@ -0,0 +1,26 @@
+<?xml version="1.0" encoding="utf-8"?>
+
+<templates id="template" inherit_id="point_of_sale.template">
+    <t t-extend="OrderReceipt">
+        <t t-jquery="div.orderlines" t-operation="before">
+            <t t-if="receipt.client">
```
Comment (pimodoo) [inline addons/l10n_in_pos/static/src/xml/pos_receipt.xml]: You should also check that the company is in india for when you are working in multicompany in different countries. So you can add in the 'export_for_printing' in point of sale module, the country on the company and check here in the t-if that country code is 'IN' before adding informations on the receipt.

**Comment by jva-odoo** [comment]
@pimodoo Done all changes.

**Comment by pimodoo** [comment]
robodoo r+ rebase-ff

**Pull request closed without merging**